oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/conditions/Validate.java

```diff
@@ -119,9 +119,9 @@ public static void checkState(final boolean expression) {
      * @param errorMessage message to use in exception
      * @throws IllegalStateException if expression is false
      */
-    public static void checkState(final boolean expression, @NotNull final String errorMessage) {
+    public static void checkState(final boolean expression, @NotNull final Object errorMessage) {
```

Comment:
This is required because in some places the string we pass is not initialized until the expression has run.